src/core/sequence_state.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
```

Comment:
The sequence state logic seems unnecessrily distributed across infer_request, triton_model, sequence_batcher. Can you instead add a SequenceStates abstraction here that along with State (perhaps renamed to SequenceState so it is not so general) can hold all that logic for creating, managing, etc. sequence state. Then, in the other places you will just hold references to SequenceStates and call into methods in that class to perform all the state related functions.